src/components/charts/useChartGroup.ts

```diff
@@ -167,17 +170,41 @@ export default function useChartGroup(chartGroupId = null): ChartGroupExports {
         setState((state) => ({ ...state, selectedInstances: getSelectedInstances(state.charts) }))
     }, [state.charts])
 
+
     async function applyFilterOnCharts(queryString: string): Promise<any> {
         try {
-            const { result: availableCharts } = await getAvailableCharts(queryString)
-            setState((state) => ({ ...state, availableCharts: mapByKey(availableCharts, 'id') }))
+            console.log("filters")
+            const { result: availableCharts } = await getAvailableCharts(queryString,state.page_offset, state.page_size)
+            setState((state) => ({ ...state, page_offset: state.page_offset + state.page_size}))
+            setState((state) => ({ ...state, availableCharts: mapByKey(availableCharts,'id') }))
+        } catch (err) {
+            showError(err)
+        } finally {
+            setState((state) => ({ ...state, loading: false }))
+        }
+    }
+
+    async function applyPaginationOnCharts( queryString: string): Promise<any> {
+
+        try{
+            const { result: availableCharts } = await getAvailableCharts(queryString,state.page_offset, state.page_size)
+            setState((state) => ({ ...state, page_offset: state.page_offset + state.page_size}))
+            setState((state) => ({ ...state, availableCharts: new Map([...state.availableCharts, ...mapByKey(availableCharts,'id')]) }))
         } catch (err) {
             showError(err)
         } finally {
             setState((state) => ({ ...state, loading: false }))
         }
+
     }
 
+    function resetPaginationOffset(): void {
```

Comment:
We Should not update the state directly Use setter to update the same. Also the function should be related